src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -53,28 +53,23 @@ public function trans($id, array $parameters = [], $domain = null, $locale = nul
             $legacy = $parameters['legacy'];
             unset($parameters['legacy']);
         }
+        $emptyParams = empty($parameters);
+        $isSprintf = !$emptyParams && $this->isSprintfString($id);
 
         if (empty($locale)) {
             $locale = null;
         }
 
-        $translated = parent::trans($id, [], $this->normalizeDomain($domain), $locale);
```

Comment:
This is still breaking legacy fallback when using placeholders.

eg. when calling `trans("This is a %message%", ['%message%' => 'bad idea'], 'Modules.Mymodule.Foobar')`

Assume the translation is not in the catalog (which should call for legacy fallback). When the translation is not found, `trans` returns the original wording without translating it. 

However, on this line, `$translated` will be "This is a bad idea" instead of "This is a %message%", because you sent the parameters and when you do, they are replaced regardless of whether the wording was translated or not.

So when we call `$this->shouldFallbackToLegacyModuleTranslation()`, it will return false instead of true, because `$id` and `$translated` should have been equal, based on the original premise stating that one of the requisites to qualify for legacy fallback is that the wording is not found on the translation catalogue, and that we can tell that because the original wording and their translations are equal. 

Therefore, my [previous comment](https://github.com/PrestaShop/PrestaShop/pull/29584#discussion_r966209517) remains true.